pandas/core/series.py

```diff
@@ -879,8 +879,15 @@ def __array__(
               dtype='datetime64[ns]')
         """
         values = self._values
-        arr = np.asarray(values, dtype=dtype)
-        if astype_is_view(values.dtype, arr.dtype):
+        if copy is None:
+            # Note: branch avoids `copy=None` for NumPy 1.x support
+            arr = np.asarray(values, dtype=dtype)
+        else:
+            arr = np.array(values, dtype=dtype, copy=copy)
+
+        if copy is True:
+            return arr
+        if copy is False or astype_is_view(values.dtype, arr.dtype):
```

Comment:
One more issue I noticed while doing the backport: for a similar case in generic.py we changed the `copy is False` to `copy is not True`, which I think we should have done here as well.

Will try to come up with a test case that would fail because of this, and do a follow-up tomorrow.

Reply:
Thanks, ouch.  This is clearly hard to get right without tests :/.

The release snippet looks fine to me.

Reply:
It also only matters for the `copy=None` case, which AFAIU is quite new, and so we didn't cover that when adding the logic here for making the resulting array read-only

Reply:
Looking closer, I think this is right?  Note that it says `copy is False or ...`.  If `copy is False` we don't have to check whether it is a view (because otherwise it would have errored)?

Reply:
Ah, that's a good point. I assume the `arr = np.array(values, dtype=dtype, copy=copy)` line above will already have errored in case of `copy=False` if a zero-copy conversion was not possible. 
So indeed the `or` is fine: if `copy=False` we know `arr` is always a view, and for `copy=None` (the one remaining case) we have `astype_is_view` to check it.

Maybe it's then in `generic.py` that we could change the `copy is not True and ..` to `copy is False or ..` (just to use a similar pattern in both cases)

Reply:
Looking once more at it, we actually already have tests for this, because we test that we get a read-only array using `np.asarray(ser)`, which essentially means passing `copy=None`. 

I expanded the test with `np.array(ser, copy=False)` to also explicitly cover the `copy=False` case (and ensure this branch setting writeable to False is covered for that case) -> https://github.com/pandas-dev/pandas/pull/60191